test/test_sparse_csr.py

```diff
@@ -1575,6 +1615,35 @@ def fn(c, b):
                 b = make_tensor(sample.args[1].shape, device=device, dtype=dtype, noncontiguous=True, requires_grad=True)
                 self.assertTrue(torch.autograd.gradcheck(fn, [c, b], fast_mode=True))
 
+            # Now test the autograd path wrt sparse inputs
+            # TODO: torch.sparse.addmm backward is not implemented for CSR
```

Comment:
What is "gradient Tensor properties"?

I've got a draft of the `sparse_mask` function for CSR that is used for sparse addmm backward in https://github.com/pytorch/pytorch/pull/68084. It's likely inefficient, we would improve on that later.

Reply:
We've been talking about figuring out an API that can allow the user to set certain properties for the gradient of a Tensor. The actual functions could look like
```
torch.sparse.sparse_gradients(torch.topk, args=(1, 2), torch.sparse_coo, *inputs)
```
We'll then need to figure out some better syntactic sugar

Reply:
It's to avoid having `use_sparse` everywhere like we do for something like [`torch.gather`](https://pytorch.org/docs/master/generated/torch.gather.html#torch.gather).

The key difference of torch.sparse.addmm vs. torch.addmm are the gradients, but we'll want to have a kinds of gradient properties, so this approach of adding another kwarg or having another function within `torch.sparse` won't scale. We need to write up a centralized proposal here.